pandas/_libs/period.pyx

```diff
@@ -771,9 +779,7 @@ cdef class _Period(object):
 
     def __add__(self, other):
         if isinstance(self, Period):
-            if isinstance(other, (timedelta, np.timedelta64,
-                                  offsets.Tick, offsets.DateOffset,
-                                  Timedelta)):
+            if isinstance(other, (timedelta, np.timedelta64, ABCDateOffset)):
```

Comment:
`Timedelta` subclasses `timedelta` and `Tick` subclasses `DateOffset`, so we can remove them from these checks.